cupyx/fallback_mode/fallback.py

```diff
@@ -365,9 +423,41 @@ def _call_numpy(func, args, kwargs):
         Result after calling func and performing data transfers.
     """
 
-    args, kwargs = _get_cupy_args(args, kwargs)
-    numpy_args, numpy_kwargs = _get_numpy_args(args, kwargs)
+    _prepare_for_numpy_dispatch(args, kwargs)
+    numpy_args, numpy_kwargs = _convert_fallback_to_numpy(args, kwargs)
     numpy_res = func(*numpy_args, **numpy_kwargs)
-    cupy_res = _get_cupy_result(numpy_res)
 
-    return _get_fallback_result(cupy_res)
+    # Sometimes reference to out(which was inplace updated) is returned
+    # ex: numpy.add, numpy.nanmean
+    # Therefore, to avoid creating separate ndarrays out is returned
+    numpy_out = numpy_kwargs.get('out', None)
+    if numpy_out is not None and numpy_out is numpy_res:
+        return kwargs.get('out')
+
+    return _convert_numpy_to_fallback(numpy_res)
+
+
+def _is_cupy_compatible(args, kwargs):
```

Comment:
If this function is for `(args, kwargs)` passed to `_RecursiveAttr.__call__`, how about define this function as a `classmethod` of `_RecursiveAttr.__call__` for readability?

Reply:
Okay, will make this as `classmethod` of `_RecursiveAttr`.

Reply:
I made it as `@staticmethod` since `cls` is not required as argument for `_is_cupy_compatible`.